relm4/src/component/sender.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2022 System76 <info@system76.com>
+// SPDX-License-Identifier: MIT or Apache-2.0
+
+use crate::{Component, Sender};
+use std::rc::Rc;
+
+/// Contain senders used by the component.
+pub type ComponentSender<C> = Rc<ComponentSenderInner<C>>;
+
+/// Contains senders used by the component.
+#[derive(Debug)]
+pub struct ComponentSenderInner<C: Component> {
+    /// Emits component inputs
+    pub input: Sender<C::Input>,
+
+    /// Emits component outputs
+    pub output: Sender<C::Output>,
+}
+
+impl<C: Component> ComponentSenderInner<C> {
+    /// Emit an input to the component.
+    pub fn input(&self, message: C::Input) {
+        self.input.send(message);
+    }
+
+    /// Emit an output to the component.
+    pub fn output(&self, message: C::Output) {
+        self.output.send(message);
+    }
+}
```

Comment:
Maybe some methods like `input_sender` and `output_sender` would be nice to access the senders.